src/porepy/numerics/ad/operator_functions.py

```diff
@@ -9,97 +16,90 @@
+    AD operators.
 
-    Implements the callable-functionality and provides abstract methods for obtaining function
-    values and the Jacobian.
-    The abstraction intends to provide means for approximating operators, where values are
-    e.g. interpolated and the Jacobian is approximated using FD.
+    Implements the call with Ad operators, creating an operator with children
+    and its operation set to
+    :attr:`~porepy.numerics.ad.operators.Operator.Operations.evaluate`.
 
-    Note:
-        One can flag the operator as ``ad_compatible``. If flagged, the AD framework passes
-        AD arrays directly to the callable ``func`` and will **not** call the abstract methods
-        for values and the Jacobian during operator parsing.
-        If for some reason one wants to flag the function as AD compatible, but still have the
-        abstract methods called, this is as of now **not** supported.
+    Provided abstract methods to implement the computation of value and Jacobian of the
+    function independently.
 
-        For now only one child class, porepy.ad.Function, flags itself always as AD compatible.
+    The abstract function itself has no arithmetic overloads, since its meaning
+    is given only by calling it using other operators. Type errors are raised if the
+    user attempts to use any overload implemented in the base class.
 
     Parameters:
-        func: callable Python object representing a (numeric) function.
-            Expected to take numerical information in some form and return numerical
-            information in the same form.
-        name: name of this instance as an AD operator
-        array_compatible (optional): If true, the callable ``func`` will be called
-            using arrays (numpy.typing.ArrayLike). Flagging this true, the user ensures
-            that the callable can work with arrays and return respectively
-            formatted output. If false, the function will be evaluated element-wise
-            (scalar input). Defaults to False.
-        ad_compatible (Optional): If true, the callable ``func`` will be called using
-            the porepy.ad.AdArray.
-
-            Note that as of now, this will effectively bypass the abstract methods
-            for generating values and the Jacobian, assuming both will be provided
-            correctly by the return value of ``func``.
-
-            Defaults to False.
+        name: Name of this instance as an AD operator.
 
     """
 
     def __init__(
         self,
-        func: Callable,
-        name: str,
-        array_compatible: bool = False,
-        ad_compatible: bool = False,
-    ):
-        ### PUBLIC
-
-        self.func: Callable = func
-        """Callable passed at instantiation"""
-
-        self.array_compatible: bool = array_compatible
-        """Indicator whether the callable can process arrays."""
-
-        super().__init__(name=name, operation=Operator.Operations.evaluate)
+        name: Optional[str] = None,
```

Comment:
The AbstractFunction has a general implemenation of the `func` member now, with get_values and get_jacobian remaining abstract.
Constructor changed to be compatible with multiple inheritance including other base clases such as the new IterativeOperator and TimeDependentOperator.